### ISSUE TITLE
bug: warpcast data not broadcast to hubs correctly

### ISSUE THREAD

**Issue by jordanlesich** [farcasterxyz/hub-monorepo#1076]
**What is the bug?**
I'm getting a 'not found' error when I try to load fName for this ID "15152". 
- The user loads on Warpcast, 
https://warpcast.com/manbir
- Using mainnet nemes hub
- Using ``@farcaster/hub-nodejs`` 
- Loads PFP and Display name, but fails when I try to get fName
![image](https://github.com/farcasterxyz/hub-monorepo/assets/53406838/71a7fb63-c2a4-490c-bf26-aa24cc24a01e)

**How can it be reproduced? (optional)**
```ts
import { getSSLHubRpcClient, UserDataType } from '@farcaster/hub-nodejs';
import dotenv from 'dotenv';

dotenv.config();

// using mainnet nemes
const hubRpcEndpoint = process.env.HUB as string;

const client = getSSLHubRpcClient(hubRpcEndpoint);

client.$.waitForReady(Date.now() + 5000, async (e) => {
  if (e) {
    console.error(`Failed to connect to ${hubRpcEndpoint}:`, e);
    process.exit(1);
  } else {
    console.log(`Connected to ${hubRpcEndpoint}`);

    // This FID works.

    // const castsResult = await client.getUserData({
    //   fid: 3,
    //   userDataType: UserDataType.FNAME,
    // });

    // This FID returns a HubError: Not Found
    const castsResult = await client.getUserData({
      fid: 15152,
      userDataType: UserDataType.FNAME,
    });

    if (castsResult.isErr()) {
      console.log('Error');
      console.log('castsResult:', castsResult);
    }

    castsResult.map((usrData) => console.log(usrData));
    client.close();
  }
});


```

**Additional context (optional)**
console error
![image](https://github.com/farcasterxyz/hub-monorepo/assets/53406838/958e7452-dc5d-44a6-82c4-60db90461312)

**Comment by varunsrin** [comment]
might be related to https://github.com/farcasterxyz/hub-monorepo/issues/1059

**Comment by varunsrin** [comment]
suspicion is that this is related to a bug in warpcast not publishing some messages correctly, @sds  and @deodad to investigate

**Comment by deodad** [comment]
Confirmed this is related to a Warpcast bug. We'll work on getting that message published and will close this ticket when that bug is resolved.

**Comment by vernonjohnson** [comment]
Any updates on this? Thought messages were republished last week but still seeing this issue

**Comment by varunsrin** [comment]
Actively working on it, but debugging has been slow due to other issues with older peers spamming messages due to a bug in 1.3.2. Working on getting everyone to upgrade so we can resolve this quickly.

**Comment by deodad** [comment]
@vernonjohnson we recently shipped a few fixes related to this issue and published any missing data

**Comment by manan19** [comment]
Confirming that the issue is fixed on my Postgres replicator.

@deodad Can you please link the exact PR fixes related to this issue to this ticket?

**Comment by deodad** [comment]
@manan19 great! the core issue here was that there were some situations where Warpcast would fail to publish the UserDataAdd message for a user, so the fixes were in the Warpcast application and not hub-monorepo

**Comment by manan19** [comment]
> note: a contributor to the problem was the missing Eth events that we've seen in some hubs and there's some in-flight work to make that more resilient, but it hasn't landed yet

@deodad Does this mean we could still have this issue recur between now and "the missing Eth events" resiliency work landing?

**Comment by varunsrin** [comment]
there are two separate issues at play: 

1. warpcast wasn't pushing events to hubs (fixed) 
2. hubs sometimes miss eth events and don't pick up data from other hubs. 

its likely that (1) caused 99% of the issues here, so we expect this to be a lot better now. 

fixes are already pushed out for (2) in 1.4.1, we're testing to see how effective it is.
